interface.go

```diff
@@ -20,6 +23,23 @@ type LegPublisher interface {
 
 // LegSubscriber is an interface for watching a published dag.
 type LegSubscriber interface {
+	Subscribe(ctx context.Context, selector ipld.Node, policy PolicyHandler) error
```

Comment:
I'm not convinced there should be a `Subscribe` method on the interface.
* subscribing to the pubsub topic and configuration there can live outside of the interface in the specific implementation constructor.
* perhaps there's a hook for `NoteUpdatedSyncState(head cid.Cid)` to short-circuit some of the subsequent announcement fetch that has been loaded through other means.

I think the main thing is that i think we should make construction and ownership over the selector be something that happens within this library, rather than something set by the client using it.
Is there something in the customization of that selector you think the client making use of legs will do better than the library itself?

Reply:
Makes sense. I don't have a strong opinion on this, so happy to do otherwise. 

My reasoning behind this was that users may want to use legs in a general way for different type of exchanges when an update for a DAG is seen. In our case we explore recursively and use a dynamic stop condition, but other subscribers may want to subscribe with a selector that exchanges all from scratch, or which fetches the node for the latest update.

If I understood correctly, your suggestion then is to set the `ExploreRecursive` selector as the one used for legs subscribers by design, and add additional config through hooks or when instantiating it, removing `Subscribe` from the interface, right?

Reply:
yeah. I think lets wait until we have clients wanting other behavior to figure out what the right interface is for that. i'd guess in general it'll be part of the constructor rather than on the interface, since it's unlikely to change dynamically within a session